### PR TITLE
Fixes #27936 - Fix changing deploy_by

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ares** [comment]
Test failures related

**Comment by xprazak2** [comment]
I know what is going on - this is not really testable without ansible. I can mock in tests that ansible is present, but I have no way of finding out whether ansible role is present, because that model does not exist.

File: app/controllers/api/v2/compliance/policies_controller.rb [hunk 1]
```diff
@@ -69,7 +69,13 @@ def create
       param_group :policy
 
       def update
-        process_response @policy.update_attributes(policy_params)
+        ::ForemanOpenscap::Policy.transaction do
```
Comment (ares) [inline app/controllers/api/v2/compliance/policies_controller.rb]: could you pleace extract this logic to model's method, somehing like

```ruby
def change_policy_deploy_type(params)
  self.class.transaction do
    ...
  end
end
```

and then reuse it in both controllers? it needs to return true/false and base on that, we'd process_response or process_success/process_error

**Comment by ares** [comment]
Thanks @xprazak2, merging

**Pull request merged**